packages/core-data/src/selectors.ts

```diff
@@ -686,6 +687,12 @@ export const __experimentalGetDirtyEntityRecords = createSelector(
 								entityConfig?.getTitle?.( entityRecord ) || '',
 							name,
 							kind,
+							hasMetaChanges: !! getEntityRecordNonTransientEdits(
```

Comment:
One important aspect to keep in mind is that not all entities define the `meta` field. Entities are defined here:

https://github.com/WordPress/gutenberg/blob/ad3094187dec15e6832c134628561a236cf0b19c/packages/core-data/src/entities.js#L23-L232

Post types and taxonomies are loaded dynamically based on another REST API request. What's important is that it isn't universally applicable. We also cover only post meta as the binding source so it might catch additional unplanned cases.

Reply:
We can iterate on it later as this selector is experimental. I'm still not convinced it's a good place to introduce the `hasMetaChanges` field, as it isn't generally applicable to all entities. It's also a derrived value that gets computed based on all other data returned from this selector.

Reply:
I was looking at the different possibilities, and I believe we could use something like [getEntityRecordsEdits](https://github.com/WordPress/gutenberg/blob/c72afdabe993565196aee28b317d001a805f1702/packages/core-data/src/selectors.ts#L762-L771) directly in the new `hasPostMetaChanges` selector we are creating. There, we can get a list of the editedEntities and with `getEntityRecordEdits` (or something similar), check if the meta property is part of that. I can take a look at that.